tornado/routing.py

```diff
@@ -242,7 +243,10 @@ def headers_received(self, start_line, headers):
             start_line=start_line, headers=headers)
 
         self.delegate = self.router.find_handler(request)
-        return self.delegate.headers_received(start_line, headers)
+        if self.delegate is not None:
+            return self.delegate.headers_received(start_line, headers)
+
+        raise _DelegateNotFoundException
```

Comment:
I had imagined this as being a default delegate that would be returned instead of an exception raised. Is there any particular reason why you went with an exception?

Reply:
I'm sure I had something in mind when I decided to use an exception) But now from the fresh perspective it seems that the implementation could be much simpler